rcl/test/rcl/test_node.cpp

```diff
@@ -437,6 +425,121 @@ TEST_F(CLASSNAME(TestNodeFixture, RMW_IMPLEMENTATION), test_rcl_node_life_cycle)
+  // Initialize logging and rosout.
+  ret = rcl_logging_configure(&context.global_arguments, &allocator);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_logging_fini()) << rcl_get_error_string().str;
+  });
+  ret = rcl_logging_rosout_init(&allocator);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_logging_rosout_fini()) << rcl_get_error_string().str;
+  });
+  // Try with invalid allocator.
+  rcl_node_options_t options_with_invalid_allocator = rcl_node_get_default_options();
+  options_with_invalid_allocator.allocator.allocate = nullptr;
+  options_with_invalid_allocator.allocator.deallocate = nullptr;
+  options_with_invalid_allocator.allocator.reallocate = nullptr;
+  ret = rcl_node_init(&node, name, namespace_, &context, &options_with_invalid_allocator);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << "Expected RCL_RET_INVALID_ARGUMENT";
+  ASSERT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+  // Try with failing allocator.
+  rcl_node_options_t options_with_failing_allocator = rcl_node_get_default_options();
+  options_with_failing_allocator.allocator.allocate = failing_malloc;
+  options_with_failing_allocator.allocator.reallocate = failing_realloc;
+  ret = rcl_node_init(&node, name, namespace_, &context, &options_with_failing_allocator);
+  EXPECT_EQ(RCL_RET_BAD_ALLOC, ret) << "Expected RCL_RET_BAD_ALLOC";
+  ASSERT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+  // Try init but force internal errors.
+  {
+    auto mock = mocking_utils::patch_and_return("lib:rcl", rmw_create_node, nullptr);
+    ret = rcl_node_init(&node, name, namespace_, &context, &options);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+  }
+
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rcl", rmw_node_get_graph_guard_condition, nullptr);
+    ret = rcl_node_init(&node, name, namespace_, &context, &options);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+  }
+
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rcl", rmw_validate_node_name, RMW_RET_ERROR);
+    ret = rcl_node_init(&node, name, namespace_, &context, &options);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+  }
+
+  {
+    auto mock = mocking_utils::patch_and_return(
+      "lib:rcl", rmw_validate_namespace, RMW_RET_ERROR);
+    ret = rcl_node_init(&node, name, namespace_, &context, &options);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+  }
+  // Try normal init but force an internal error on fini.
+  {
+    ret = rcl_node_init(&node, name, namespace_, &context, &options);
+    EXPECT_EQ(RCL_RET_OK, ret);
+    auto mock = mocking_utils::inject_on_return("lib:rcl", rmw_destroy_node, RMW_RET_ERROR);
+    ret = rcl_node_fini(&node);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+  }
+
+  // Battle test node init/fini.
```

Comment:
In this test, only init is the one that could fail due to fault injection, right? in that case 
```suggestion
  // Battle test node init.
```

Reply:
That is correct! See b400aeea833a7cdb2b2856aea85ed82e3b43beea.